ext/imap/php_imap.c

```diff
@@ -790,11 +802,22 @@ PHP_FUNCTION(imap_open)
 	 * the CL_EXPUNGE flag which will expunge when the mailbox is closed (be that manually, or via the
 	 * IMAP\Connection object being destroyed naturally at the end of the PHP script */
 	if (flags && ((flags & ~(OP_READONLY | OP_ANONYMOUS | OP_HALFOPEN | PHP_EXPUNGE | OP_DEBUG | OP_SHORTCACHE
-	 		| OP_SILENT | OP_PROTOTYPE | OP_SECURE)) != 0)) {
+	 		| OP_SILENT | OP_PROTOTYPE | OP_SECURE | PHP_XOAUTH2)) != 0)) {
 		zend_argument_value_error(4, "must be a bitmask of the OP_* constants, and CL_EXPUNGE");
 		RETURN_THROWS();
 	}
 
+
+#ifdef HAVE_IMAP_XOAUTH2
+	/* We only enable the XOAUTH2 authenticator if explicitly requested */
+	mail_parameters(NIL, (flags & PHP_XOAUTH2) ? ENABLE_AUTHENTICATOR : DISABLE_AUTHENTICATOR, (void *)"XOAUTH2");
+#else
+	if (flags & PHP_XOAUTH2) {
+		zend_argument_value_error(4, "not compiled with XOAUTH2 support");
+		RETURN_THROWS();
```

Comment:
Not sure if this should be a ValueError or an E_WARNING with a false return

Reply:
Returning false with a warning seemed to be the solution of choice in most of the other few similar checks I could find, so I changed it.